src/main/java/fr/pickaria/pterodactylpoweraction/ConnectionListener.java

```diff
@@ -172,7 +179,10 @@ private void scheduleServerShutdown(Player player) {
     }
 
     private void scheduleServerShutdown(RegisteredServer registeredServer) {
-        shutdownManager.scheduleShutdown(registeredServer);
+        String serverName = registeredServer.getServerInfo().getName();
+        if (configurationLoader.getConfiguration().getAllServers().contains(serverName)) {
+            shutdownManager.scheduleShutdown(registeredServer);
+        }
```

Comment:
After adding the new private method, you can modify this here:

```suggestion
        if (this.isManagedServer(registeredServer)) {
            shutdownManager.scheduleShutdown(registeredServer);
        }
```